pyfar/dsp/dsp.py

```diff
@@ -850,3 +851,57 @@ def _cross_fade(first, second, indices):
     result = first * window_first + second * window_second
 
     return result
+
+
+def time_shift(signal, shift, unit='samples'):
+    """Apply a time-shift to a signal.
+
+    The shift is performed as a cyclic shift on the time axis, potentially
+    resulting in non-causal signals for negative shift values.
+
+    Parameters
+    ----------
+    signal : Signal
+        The signal to be shifted
+    shift : int, float
+        The time-shift value. A positive value will result in right shift on
+        the time axis (delaying of the signal), whereas a negative value
+        yields a left shift on the time axis (non-causal shift to a earlier
+        time).
```

Comment:
It is not clear, that individual shifts for the channels can be used. Is the resulting error for a mismatch of `cshape` and `shift.shape` easy to understand?

Reply:
Indeed a good point, I've added documentation on this and added an example